### PR TITLE
Silence JS warnings for CE 2

### PR DESCRIPTION
**Pull request by danicheg**
Even if we don't use fatal warnings, it, all the same, strikes the eye on every compile, CI run.
```
The global execution context in Scala.js is based on JS Promises (microtasks).
[warn] Using it may prevent macrotasks (I/O, timers, UI rendering) from running reliably.
[warn] 
[warn] Unfortunately, there is no way with ECMAScript only to implement a performant
[warn] macrotask execution context (and hence Scala.js core does not contain one).
[warn] 
[warn] We recommend you use: https://github.com/scala-js/scala-js-macrotask-executor
[warn] Please refer to the README.md of that project for more details regarding
[warn] microtask vs. macrotask execution contexts.
[warn] 
[warn] If you do not care about macrotask fairness, you can silence this warning by:
[warn] - Adding @nowarn("cat=other") (Scala >= 2.13.x only)
[warn] - Setting the -P:scalajs:nowarnGlobalExecutionContext compiler option
[warn] - Using scala.scalajs.concurrent.JSExecutionContext.queue
[warn]   (the implementation of ExecutionContext.global in Scala.js) directly.
[warn] 
[warn] If you do not care about performance, you can use
[warn] scala.scalajs.concurrent.QueueExecutionContext.timeouts().
[warn] It is based on setTimeout which makes it fair but slow (due to clamping)
[warn]                 
[warn]     IO.contextShift(ExecutionContext.global)
```

### CODE REVIEW

**Comment by armanbilge** [comment]
Hmm, why not use the `MacrotaskExecutor` like the warning suggests?

**Comment by danicheg** [comment]
Damn it, Scala 3 doesn't contain `-P:scalajs:nowarnGlobalExecutionContext` option :(

**Comment by armanbilge** [comment]
FWIW the `MacrotaskExecutor` project was extracted from CE, is maintained by Daniel and I, and is the default in CE3.

**Comment by danicheg** [comment]
@armanbilge so, I've followed your (and compiler's one) advice to use `scala-js-macrotask-executor` for CE2. If CE3 uses it as a default, I think it'd be fine for CE2 as well.

File: ce2/shared/src/main/scala/munit/CatsEffectSuite.scala [hunk 1]
```diff
@@ -27,11 +27,13 @@ abstract class CatsEffectSuite
     with CatsEffectFixtures
     with CatsEffectFunFixtures {
 
+  private val ec: ExecutionContext = suiteExecutionContext
+
   implicit def munitContextShift: ContextShift[IO] =
-    IO.contextShift(ExecutionContext.global)
```
Comment (armanbilge) [inline ce2/shared/src/main/scala/munit/CatsEffectSuite.scala:31]: Is it possible to use [`munitExecutionContext`](https://www.javadoc.io/static/org.scalameta/munit_3/0.7.29/api/munit/FunSuite.html#munitExecutionContext-0) instead?

Comment (danicheg) [reply ce2/shared/src/main/scala/munit/CatsEffectSuite.scala:31]: That was intentional https://github.com/typelevel/munit-cats-effect/pull/65

Comment (armanbilge) [reply ce2/shared/src/main/scala/munit/CatsEffectSuite.scala:31]: Wow, munit uses parisitic EC ?? thanks for the pointer.

**Review by armanbilge — APPROVED** [review]
Thanks, LGTM 👍

**Pull request merged**